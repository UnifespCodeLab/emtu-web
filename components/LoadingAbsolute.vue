<template>
  <v-overlay>
    <v-progress-circular
      :size="50"
      color="primary"
      indeterminate
    />
  </v-overlay>
</template>
