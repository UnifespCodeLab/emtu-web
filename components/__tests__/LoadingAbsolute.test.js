import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import LoadingAbsolute from '~/components/LoadingAbsolute'

const localVue = createLocalVue()
localVue.use(Vuetify);

describe('Components / LoadingAbsolute', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()

    wrapper = mount(LoadingAbsolute, {
      localVue,
      vuetify
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has a loading icon with the correct props', () => {
    const icon = wrapper.findComponent({ name: 'v-progress-circular' })
    expect(icon.exists()).toBeTruthy()
    expect(icon.props().color).toBe('primary')
    expect(icon.attributes('style')).toContain('height: 50px; width: 50px;')
  })
})
