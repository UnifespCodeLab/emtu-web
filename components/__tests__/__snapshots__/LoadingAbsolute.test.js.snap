// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / LoadingAbsolute matches the snapshot 1`] = `
<div
  class="v-overlay v-overlay--active theme--dark"
  style="z-index: 5;"
>
  <div
    class="v-overlay__scrim"
    style="opacity: 0.46; background-color: rgb(33, 33, 33); border-color: #212121;"
  />
  <div
    class="v-overlay__content"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      class="v-progress-circular v-progress-circular--visible v-progress-circular--indeterminate primary--text"
      role="progressbar"
      style="height: 50px; width: 50px;"
    >
      <svg
        style="transform: rotate(0deg);"
        viewBox="21.73913043478261 21.73913043478261 43.47826086956522 43.47826086956522"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="v-progress-circular__overlay"
          cx="43.47826086956522"
          cy="43.47826086956522"
          fill="transparent"
          r="20"
          stroke-dasharray="125.664"
          stroke-dashoffset="125.66370614359172px"
          stroke-width="3.4782608695652177"
        />
      </svg>
      <div
        class="v-progress-circular__info"
      />
    </div>
  </div>
</div>
`;
